src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/CommandSet.cs

```
@@ -62,12 +60,12 @@ public CommandSet(ISite site)
     {
         this.site = site;
 
-        _eventService = (IEventHandlerService)site.GetService(typeof(IEventHandlerService));
+        _eventService = site.GetService<IEventHandlerService>()!;
         Debug.Assert(_eventService is not null, "Command set must have the event service.  Is command set being initialized too early?");
```

💡 Consider creating a `GetRequiredService` extension method for this pattern.

https://github.com/dotnet/roslyn/blob/0caa880a161057559d1dcaad890e4053f520136e/src/Workspaces/Core/Portable/Workspace/Host/HostWorkspaceServices.cs#L43-L57

---

It raises the question of throwing new exceptions, so I'll let the team decide what they want to do with this one

---

imo I would much rather have it throw an exception like this rather than a null reference, but I will check with team on this among some of the other questions about raising a more specific exception

---

+1 for a specific exception

---

Done. I used the already existing `GetRequiredService` and deleted the assert